### DISCUSSION TITLE
How to account for days off in my Teplater + Daily Notes setup up

### DISCUSSION THREAD

**Discussion by cyberyanka** [SilentVoid13/Templater#718] [Help]
I have [a function](https://github.com/SilentVoid13/Templater/discussions/240) to determine which template for a new daily note to use depending on which day of the week it is.

I have two templates: for a weekday and for a weekend.

Now I'm about to go on vacation and I want my daily notes to behave accordingly.

My idea was to place all dates in a note "Holidays" (each date on a new line) and have my function parse it before creating a new daily note.

I tried to do it with available functions `tp.file`, but it seems like this module is for interacting with template files or target files and isn't suitable for retrieving data from some file that isn't a template nor a target for insertion of a template.

Is it possible to do? How?

And if not, what other way to do something like it?

My function:

```
function scheduleTemplate(tp) {
    switch (parseInt(moment(tp.file.title).format('d'))) {
        case 6:
        case 0:
            return tp.file.include("[[Weekend]]")
        default:
            return tp.file.include("[[Weekday]]")
    }
}

module.exports = scheduleTemplate;
```

My Holidays note:

```
2022-08-15
2022-08-16
2022-08-17
2022-08-18
2022-08-19
```

**Comment by welpdx** [comment]
It sounds like you just want to grab the content of another note add it into current file without parsing any templater command. 

What you can do is use `await this app.vault.read(tFile)`. This command will get the text content of a note via it's `tfile`.


So something like this:
```js
<%*
// get tfile
var holidayNote = await this.app.vault.getAbstractFileByPath("foo/bar/holiday.md")
// add content of note into current note
tR+=  "\n" + await this.app.vault.read(holidayNote) + "\n"
%>
```
